ex2_hal/iris/hardware_interface/source/iris_bootloader_cmds.c

```diff
@@ -337,3 +302,46 @@ int iris_mass_erase_flash() {
     return ret;
 }
 
+uint16_t get_file_size(int32_t fptr) {
+    REDSTAT file_stat;
+
+    red_fstat(fptr, &file_stat);
+
+    return (uint16_t)file_stat.st_size;
```

Comment:
I'm pretty sure st_size is way bigger than an int16. This kind of cast is dangerous

Reply:
Using uint32_t now :+1: